options/hooks/useFetchKeywordSearchData.js

```diff
@@ -17,11 +18,21 @@ const useFetchKeywordSearchList = (
       const fetchEncodedUrlList = bookmarkList.map((bookmark) => {
         const encodedUrl = encodeURIComponent(bookmark.url);
 
-        if (keyword) {
+        if (keyword && userSelectValue === "키워드 검색") {
           setHasSearchResult(true);
           return fetch(
             `${SERVER_URL}/crawl/${encodedUrl}/search?keyword=${keyword}`
           );
+        } else if (keyword && userSelectValue === "제목 검색") {
+          setHasSearchResult(true);
+          return fetch(
+            `${SERVER_URL}/crawl/title/${encodedUrl}/search?keyword=${keyword}`
+          );
+        } else if (keyword && userSelectValue === "제목 + 키워드 검색") {
+          setHasSearchResult(true);
+          return fetch(
+            `${SERVER_URL}/crawl/all/${encodedUrl}/search?keyword=${keyword}`
+          );
```

Comment:
각 케이스 별로 나누어 fetch보낼 URL을 나누는 좋은 로직이라고 생각합니다.
이미 좋은 로직이지만 그래도 조금 더 개선할 방법을 찾아본다면 저는 이런 방법을 추천드립니다.
```suggestion
if (keyword) {
  let url = "";

  if (userSelectValue === "키워드 검색") {
    url = `${SERVER_URL}/crawl/${encodedUrl}/search?keyword=${keyword}`;
  }
  if (userSelectValue === "제목 검색") {
    url = `${SERVER_URL}/crawl/title/${encodedUrl}/search?keyword=${keyword}`;
  }
  if (userSelectValue === "제목 + 키워드 검색") {
    url = `${SERVER_URL}/crawl/all/${encodedUrl}/search?keyword=${keyword}`;
  }

  return fetch(url);
}
```

Reply:
처음 조건으로 keyword 을 거친 뒤 이후 조건을 거쳐서 최종으로 return fetch(url); 받는 로직 이해하였습니다.
코드의 가독성이 더 올라 간 것  같다고 느꼈습니다. 전달 주신 피드백 참고하여 적용할 수 있도록 하겠습니다. 좋은 피드백 감사합니다,